folsom/src/main/java/com/spotify/folsom/client/ascii/AsciiMemcacheDecoder.java

```diff
@@ -179,6 +183,7 @@ protected void decode(final ChannelHandlerContext ctx, final ByteBuf buf, final
             this.value = new byte[size];
             this.valueOffset = 0;
             this.cas = cas;
+            this.flags = new Flags(flags);
```

Comment:
If we store this as a primitive int, we can avoid some unnecessary garbage creation